### ISSUE TITLE
Allow WorkEstimatorConfig to be configured

### ISSUE THREAD

**Issue by mengqiy** [kubernetes/kubernetes#128628]
### What would you like to be added?

Allow WorkEstimatorConfig especially `MaximumSeatsLimit` to be configured.
Currently it's hard-coded to 10 and there's no way for a cluster admin to configure it.

https://github.com/kubernetes/kubernetes/blob/e2bf630940946df5bc161d224e4a9b2e191a3b2e/staging/src/k8s.io/apiserver/pkg/server/config.go#L1010-L1012
https://github.com/kubernetes/kubernetes/blob/e2bf630940946df5bc161d224e4a9b2e191a3b2e/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/config.go#L27

### Why is this needed?

It's fairly common that a list request is more than 10X more expensive than a single object operation e.g. get, update, create.
This underestimation can cause APIServer to admit more list requests and increase OOM risk. We don't panelize list requests enough.
If there's a knob to configure this maximumSeatsLimit to higher like 20 or even higher, it can help the APIServer to panelize and throttle expensive list request.

**Comment by cyclinder** [comment]
Yes I think it's reasonable, Do we need only `MaximumSeatsLimit`or all `WorkEstimatorConfig` to be configured?

**Comment by mengqiy** [comment]
Configuring `MaximumSeatsLimit` is good enough for my use case.

**Comment by MikeSpreitzer** [comment]
I presume you have looked at #111222 and #118601.

Are we talking about a new API object to carry this configuration? And it would start out with just one field? What is the plan for what happens later if we decide to add more fields? There is currently some internal structure in the existing WorkEstimatorConfig struct; would that be followed or would the API object be flat?

**Comment by mengqiy** [comment]
@MikeSpreitzer Does the above answer make sense to you? If so, we can start to draft PR.

**Comment by yue9944882** [comment]
A quicker close equivalent configuration here is to increase the handsize. e.g. 10->20 the max seats is effectively similar to increasing the handsize from 6 to 12. (The matrix [here](https://github.com/kubernetes/kubernetes/pull/118601#issuecomment-1631034011) roughly shows their relation). If we want to make this configurable, I think we might need to adjust the another hard-coded `15%` percentage [here](https://github.com/kubernetes/kubernetes/blob/e2bf630940946df5bc161d224e4a9b2e191a3b2e/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go#L63) accordingly which is previously determined by this 10 limit.

**Comment by MikeSpreitzer** [comment]
It is a smaller and easier change if it does not require new kinds of API objects. Go ahead and draft a PR.
